lib/elixir/test/elixir/kernel/sigils_test.exs

```diff
@@ -121,6 +187,31 @@ bar) in ["foo\\\nbar", "foo\\\r\nbar"]
     assert ~W(Foo.Bar.Baz)a == [:"Foo.Bar.Baz"]
   end
 
+  @compile {:no_warn_undefined, Kernel.SigilsTest.TrailingComma}
+
+  test "sigil w/W warns on trailing comma at compile time, not runtime" do
+    for sigil <- ~w(w W),
+        modifier <- ~w(a s c) do
+      assert capture_err(fn ->
```

Comment:
If we use capture_err the test cannot be `async`. My suggestion is to move it to kernel/warnings_test. We also have conveniences there to make writing the test easy.